File: mooc-2024/part-8/1-objects-and-methods/01-the-smallest-average-list/main.py
person1 = {"name": "Mary", "result1": 2, "result2": 3, "result3": 3}
person2 = {"name": "Gary", "result1": 5, "result2": 1, "result3": 8}
person3 = {"name": "Larry", "result1": 3, "result2": 1, "result3": 1}

print(smallest_average([person1, person2, person3])

def smallest_average(persons):
    avgs = []

    for idx, person in enumerate(persons):
        avgs[idx] = (person['result1'] + person['result2'] + person['result3']) / 3

    return persons[avgs.index(min(avgs))]

File: mooc-2024/part-8/1-objects-and-methods/01-the-smallest-average-list/main.c
#include <stdio.h>

struct Result {
    char *name;
    int result1;
    int result2;
    int result3;
} results[] = {
    "Mary", 2, 3, 3,
    "Gary", 5, 1, 8,
    "Larry", 3, 1 ,1
};

void print_result(struct Result *);
struct Result *find_min_avg(struct Result *, int);

int main() {
    struct Result *rp = find_min_avg(results, 3);
    print_result(rp);
}

struct Result *find_min_avg(struct Result results[], int resultsSize) {
    int min_idx = 0;
    float min = (results[0].result1+results[0].result3+results[0].result3) / 3;

    for (int i = 1; i < resultsSize; i++) {
        int avg = (results[i].result1+results[i].result3+results[i].result3) / 3;
        if (avg < min) {
            min_idx = i;
        }
    }

    return &results[min_idx];
}

void print_result(struct Result *result) {
    printf("%s: result1: %i, result2: %i, result3: %i\n", result->name, result->result1, result->result2, result->result3);
}